### ISSUE TITLE
Uses of panic! in script_thread

### ISSUE THREAD

**Issue by asajeffrey** [servo/servo#11685]
There are uses of panic! in script_thread.rs which have race conditions:

```
        panic!("resize sent to nonexistent pipeline");
        panic!("Page rect message sent to nonexistent pipeline");
        panic!("Set scroll state message message sent to nonexistent pipeline: {:?}", id);
        panic!("freeze sent to nonexistent pipeline");
        panic!("thaw sent to nonexistent pipeline");
```

the pipeline might have been torn down before the message was sent. These should probably all be warn!

cc @Ms2ger

**Comment by timvandermeij** [comment]
Fixed by #11695?

**Comment by ConnorGBrewster** [comment]
@timvandermeij not yet, there still some uses of `panic!` in the script thread.

**Comment by asajeffrey** [comment]
The one remaining `panic!` which has a race condition is:
```
panic!("Layout attached to empty script thread."),
```
which should be fixed by #14885.
